mifosng-android/src/main/java/com/mifos/api/datamanager/DataManagerSearch.java

```diff
@@ -17,14 +25,55 @@
 public class DataManagerSearch {
 
     public final BaseApiManager baseApiManager;
+    public final DatabaseHelperClient mDatabaseHelperClient;
+    public final DatabaseHelperGroups mDatabaseHelperGroup;
+    public final DatabaseHelperCenter mDatabaseHelperCenter;
+    public final DatabaseHelperLoan mDatabaseHelperLoan;
+    public final DatabaseHelperSavings mDatabaseHelperSaving;
 
     @Inject
-    public DataManagerSearch(BaseApiManager baseApiManager) {
+    public DataManagerSearch(BaseApiManager baseApiManager,
+                             DatabaseHelperClient databaseHelperClient,
+                             DatabaseHelperGroups databaseHelperGroup,
+                             DatabaseHelperCenter databaseHelperCenter,
+                             DatabaseHelperLoan databaseHelperLoan,
+                             DatabaseHelperSavings databaseHelperSaving) {
         this.baseApiManager = baseApiManager;
+        mDatabaseHelperClient = databaseHelperClient;
+        mDatabaseHelperGroup = databaseHelperGroup;
+        mDatabaseHelperCenter = databaseHelperCenter;
+        mDatabaseHelperLoan = databaseHelperLoan;
+        mDatabaseHelperSaving = databaseHelperSaving;
     }
 
-    public Observable<List<SearchedEntity>> searchResources(String query, String resources,
-                                                            Boolean exactMatch) {
-        return baseApiManager.getSearchApi().searchResources(query, resources, exactMatch);
+    public Observable<List<SearchedEntity>> searchResources(final String query, String resources,
+                                                            final Boolean exactMatch) {
+        switch (PrefManager.getUserStatus()) {
+            case 0:
+                return baseApiManager.getSearchApi().searchResources(query, resources, exactMatch);
+            case 1:
+                if (resources.equals(Constants.ENTITY_TYPE_CLIENTS)) {
+                    return mDatabaseHelperClient.readAllClientsByQuery(query, exactMatch);
+                } else if (resources.equals(Constants.ENTITY_TYPE_GROUPS)) {
+                    return
+                            mDatabaseHelperGroup.readAllGroupsByQuery(query, exactMatch)
+                                    .mergeWith(mDatabaseHelperCenter.readAllCentersByQuery(query,
```

Comment:
Here you go @mayank-kgp https://github.com/amitshekhariitbhu/RxJava2-Android-Samples/blob/master/app/src/main/java/com/rxjava2/android/samples/ui/operators/MergeExampleActivity.java#L54
It will merge the two observables of the same type. I think is same in both Rxjava 1 and Rxjava 2